holmes/core/tool_calling_llm.py

```
@@ -1325,6 +1326,17 @@ def call_stream(
 
                     tool_call_result: ToolCallResult = future.result()
 
+                    # Fold in LLM usage incurred inside the tool itself (e.g. a
+                    # sub-agent delegation) so reported costs/tokens stay honest.
+                    if tool_call_result.result.llm_usage:
+                        try:
+                            stats += RequestStats(**tool_call_result.result.llm_usage)
+                        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)